### DISCUSSION TITLE
Funkstatistik-Menü: Beutung der dortigen Angaben und beim DPL

### DISCUSSION THREAD

**Discussion by Fokker-1** [hoylabs/OpenDTU-OnBattery#1352] [Q&A]
Guten Tag,

ich habe aktuell erhebliche Probleme mit sporadischen Abbrüchen beim Betrieb meiner Nulleinspeisung mit Akku über openDTU on Battery und HMS-1000-2t.

Die "Funkstatistik" (danke für die Implementierung, echt prima) kann hier sehr hilfreich sein bei der Ursachenforschung, allerdings habe ich da einige Verständnisprobleme mit den Begrifflichkeiten, liegt sicher an mir, bin da nicht so der Profi.

Das sieht so aus:

![oDTU Fusion WiFi-Reset_Fehler_2024-10-16](https://github.com/user-attachments/assets/fc0aa9a3-535a-46e1-a223-a792168a1b60)

Ich liste das jetzt einfach mal mit meinen Interpretationen auf:

1) "Gesendete Anfragen:
Die Anzahl der Anfragen, die die DTU an den WR gesendet hat.

2) Empfang erfolgreich:
Der WR sendet an die DTU die Bestätigung für einen erfolgreichen, also formal einwandfreien Empfang der DTU-Anfrage zurück.

3) Empfang Fehler: Nichts empfangen:
Die DTU erwartet aufgrund ihrer Anfrage eine Antwort des W., allerdings kommt darauf keine Antwort an die DTU in einem Zeitraum von xxx sec zurück.

4) Empfang Fehler: Teilweise empfangen:
Die DTU erhält vom WR eine Antwort auf ihre Abfrage, allerdings erfolgt diese nur "teilweise".
Aber was bedeutet "teilweise" genau?

5) Empfang Fehler: Beschädigt empfangen:
Was genau bedeutet "beschädigt" und was ist der Unterschied zu "teilweise"?

6) Gesendete Fragment Wiederanforderungen
Was genaau wird als "Fragment" bezeichnet. Hat das der WR gesendet oder die DTU.
Was ist der Unterschied zwischen: Fragment / beschädigt / teilweise?

Mag ja sein, dass solche Interpretationen bei den Profis hier nur Schmunzeln erzeugen - aber nicht jeder kann ja Profi sein.
Also, ich hoffe (trotzdem) auf Eure Hilfe. Vielleicht gelingt es mir ja damit, das bestehende Problem mit der Nulleinspeisung zu lösen.

Schon mal danke im Voraus und Grüße
Werner

**Comment by Fokker-1** [comment]
... noch eine ergänzende Frage:

In der Konfiguration beim Dynamic Power Limiter (DPL) der openDTU on Battery gibt es unter "Wechselrichter" eine Auswahl: "Wechselrichter wird von Solarmodulen gespeist":


![WR durch Solarmodule](https://github.com/user-attachments/assets/316c608b-8641-472e-919c-63271b19e5ff)


Was bewirkt das funktionell bei der DTU, wenn man das aktiviert oder nicht?
Ich habe nichts dazu gefunden.

Danke und Grüße
Werner

**Comment by stefan123t** [comment]
@Fokker-1 die DTU und der WR kommunizieren mit Kommandos und Antworten. Ein Kommando ist i.d.R. auch nur ein Paket lang. Das quittiert der WR meist mit einer Antwort, die auch nur aus einem Paket bestehen kann. Meist aber sind die Antworten zwei, drei oder mehr Pakete/Fragmente lang. Das liegt daran dass in ein Paket neben MainCmd, WR/DTU Adressen, Paketzähler und SubCmd sowie der abschließenden CRC8 Checksumme eigentlich nur 12 Bytes Nutzdaten passen. Siehe z.B. auch das unten angegebene Protokoll eines sehr häufigen RealTimeRunData_Debug Kommandos, welches eben die ganzen aktuellen Daten vom Wechselrichter anfragt, die OpenDTU dann anzeigt / weiterleitet.
Wenn die DTU also Pakete empfängt und die Prüfsumme oder der Paketzähler deuten an, dass ein Paket fehlerhaft ist oder komplett fehlt weil es nicht empfangen wurde, dann kann die DTU diese Pakete innerhalb eines gewissen Zeitraums erneut vom WR anfragen. Nach einigen erneuten Versuchen die fehlenden Fragmente / Pakete für die Antwortnachricht zu bekommen gibt die DTU aber auf und versucht es beim nächsten Mal einfach wieder von vorne.
Du kannst Dir die Kommandos und die Antwortpakete des WR übrigens auf der Serial Console ansehen bzw. hier im Wiki die dazu [bekannten Informationen zum Protokoll](https://github.com/lumapu/ahoy/wiki/Protocol#realtimerundata_debug--0x0b) nachlesen.

**Reply by Fokker-1** [reply]
Guten Morgen, Stefan123,

ganz herzlichen Dank für die detaillierte und für mich sehr hilfreiche Information, super.
Das hat mich im grundlegenden Verständnis ein Stück weiter gebracht.

Die Probleme bestehen auch weiterhin, wie schon in den letzten 2 Wochen. Das System unterbricht die Rückspeisung aus dem Akku, ohne dass ich dazu einen erkennbaren Grund sehe. Insbesondere auch nachts, wenn die Leistungsaufnahme im Haus (bis auf das Schalten des Kühlschranks) praktisch konstant ist.

In der DTU finde ich keine Möglichkeit, zu sehen, wie oft und zu welchem Zeitpunkt das passiert. Immer verbunden aber ist es mit dieser mysteriösen Blinkfrequenz "LED grün alle 4 sec", die ja nach Auskunft der Hoymiles-Hotline nicht bekannt ist. Ich habe gestern die gleiche Frage noch einmal dahin gestellt, vielleicht weiß man da ja nach einigen Monaten nun mehr.

Ich habe eine Möglichkeit gefunden, dieses Ein- / Ausschalten der Einspeisung aus dem Akku zu dokumentieren.
Dazu habe ich in Home Assistant die folgenden Daten ausgewertet:


![Screenshot_20241028_074230_Firefox](https://github.com/user-attachments/assets/2c5bb6e8-c72d-4ab5-9eab-d4f89d298001)

Die obere Darstellung zeigt, wann die Einspeisung aktiv ist und wann sie ausgeschaltet wurde ON / OFF. Der gelbe Bereich ist ON, der graue OFF.

Die untere Kurve zeigt die kumulierte Tageseinspeisung (yield day) der Leistung.

Beide Darstellungen zeigen übereinstimmend die Unterbrechungen während der letzten Nacht, die auch mit den letzten Tagen vergleichbar sind.

Immer dann, wenn in der unteren Kurve der abrupte Abbruch auf Null kam, ist in der DTU ein Reset erfolgt, das war sieben Male. In der unteren Kurve ist die Auflösung beschränkt, solche Resets sehr kurz hintereinander sieht man da kaum, aber in der oberen Darstellung.

Einen Test werde ich noch machen: einen String abtrennen und den WR nur mit einem String am Akku betreiben.

Warum?
Es ist gelegentlich mal zu lesen, dass das Paralleleschalten der beiden Strings an einem Akku Probleme machen könnte. Aber nichts Genaues weiß man wohl nicht. Da die Minusleitungen der beiden Strings im WR verbunden sind, muss man bei der externen Verkabelung sehr darauf auchten, dass auf der internen Minus-Verbindung keine Ausgleichsströme fließen. Das ist bei mir gegeben, denn der Leistungsunterschied zwischen den beiden Strings liegt bei nur 0,5 W.

Tja, was nun weiter, mit solchen Macken will ich das System eigentlich nicht betreiben.
Schade, dass hier wohl offensichtlich kein Mitleser eine solche Kombi aus DTU on Batt und einem HMS hat, ein Info-Austausch wäre da natürlich extrem hilfreich. Aber vielleicht habe ich ja doch noch Glüdk ...

Noch mal vielen Dank für Deinen Hilfe und Grüße
Werner

**Reply by Nailuj1** [reply]
Hallo lieber Werner.
Ich habe das gleich Problem. Hast du schon eine Lösung gefunden?

**Comment by Fokker-1** [comment]
Hallo Nailuj1,

nein, das konnte ich trotz vieler Aktionen nicht lösen.
Die genaue Ursache blieb im Dunklen, im Web kann man auch hauptsächlich Vermutungen lesen, warum die Kombi HMS-1000 am Akku 24 V immer wieder instabil läuft. Offensichtlich "mögen" die beiden sich nicht so richtig ...

Ich hatte dann das Konzept geändert und verwende seit längerer Zeit als Inverter statt eines HMS-1000 nun einen Lumentree (SUN) mit einem Trucki-Stick T2SG und den 24 V-Akku.

Diese Kombi läuft problemlos.
Leider wird der Akku-Strom bei Einspeisung nicht angezeigt - na ja, da kann ich mit leben.

Grüße
Werner
